extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/internal/discovery/WeatherUndergroundDiscoveryService.java

```diff
@@ -43,11 +41,9 @@
  *
  * @author Laurent Garnier - Initial Contribution
  */
-@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.weatherunderground")
```

Comment:
Since the discovery service is not a component anymore (you create it in the factory with all dependencies in the constructor) you can remove a whole bunch of methods here:

activate, modified, and all set* unset* methods which are unused now.

Reply:
done